engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java

```diff
@@ -82,5 +80,122 @@ private List<Map<String, Object>> extractReferences(ComponentUsageDetails usage,
+        return service.getComponentUsageDetails(componentCode, listRequest);
+    }
+
+    @Override
+    public ComponentDeleteResponse deleteInternalComponents(List<ComponentDeleteRequestRow> components) {
+        ComponentDeleteResponse response = new ComponentDeleteResponse();
+        log.debug("request deletion of components '{}'", components);
+        try {
+            ComponentDeletionStep componentDeletionStep = new ComponentDeletionStep(false);
+            response.setStatus(ComponentDeleteResponse.STATUS_SUCCESS);
+            components.forEach(componentDeleteRequestRow -> {
+                String type = componentDeleteRequestRow.getType();
+                String code = componentDeleteRequestRow.getCode();
+                log.debug("Type '{}', Object code '{}'", type, code);
+                services.stream()
+                        .filter(s -> type.equalsIgnoreCase(s.getObjectType())).findFirst()
+                        .ifPresentOrElse(service -> {
+                                    ComponentDeleteResponseRow responseRow = null;
+                                    try {
+                                        responseRow = service.getComponentDto(code)
+                                                .map(c -> this.extractUsageDetails(code, service))
+                                                .map(usage -> check(components, code, usage))
+                                                .map(c -> delete(c, type, service))
+                                                .orElseGet(() -> ComponentDeleteResponseRow.builder().code(code).type(type)
+                                                        .status(ComponentDeleteResponse.STATUS_FAILURE).build());
+                                    } catch (Exception e) {
+                                        log.warn("Generic error when deleting element with code: '{}', type: '{}'",
+                                                code, type, e);
+                                        responseRow = ComponentDeleteResponseRow.builder().code(code).type(type)
+                                                .status(ComponentDeleteResponse.STATUS_FAILURE).build();
+
+                                    }
+
+                                    if (responseRow.getStatus().equals(ComponentDeleteResponse.STATUS_FAILURE)) {
+                                        response.setStatus(
+                                                computeOverallStatus(componentDeletionStep.isSuccessful()));
+                                    } else {
+                                        componentDeletionStep.setSuccessful(true);
+                                        if(!response.getStatus().equals(ComponentDeleteResponse.STATUS_SUCCESS)){
+                                            response.setStatus(computeOverallStatus(componentDeletionStep.isSuccessful()));
+                                        }
+                                    }
```

Comment:
this code is really complicated without the need, @mirkoPan pls:
- put here an if only to set `componentDeletionStep` with true value
- encapsulate all the logic to compute the overall status inside the `computeOverallStatus` method

Reply:
done